packages/mobile/src/hooks/useDeferredElement.tsx

```diff
@@ -0,0 +1,35 @@
+import { useState, useCallback } from 'react'
+
+import { InView } from 'react-native-intersection-observer'
+
+/** Hook that provides a ref and inView state for React Native intersection observer.
+ * Uses the InView component internally to handle intersection detection.
+ */
+export const useDeferredElement = (args?: {
+  threshold?: number
+  rootMargin?: string
+  triggerOnce?: boolean
+  fallbackInView?: boolean
+}) => {
+  const [inView, setInView] = useState(false)
+
+  const handleChange = useCallback((isInView: boolean) => {
+    setInView(isInView)
+  }, [])
+
+  const InViewWrapper = useCallback(
+    ({ children }: { children: React.ReactNode }) => {
+      return (
+        <InView triggerOnce={args?.triggerOnce ?? true} onChange={handleChange}>
```

Comment:
oh interesting. I tried to do this in web but adding an additional element around everything messed with flex layouts. So I had to go the route of attaching the `ref` to the child directly.

Reply:
yeah good point, going to drop this prop. InView also has an `as` prop so i may do `as={ExploreSection}` to further reduce the wrapping